ghproxy/ghcache/ghcache.go

```diff
@@ -170,6 +197,10 @@ func (u upstreamTransport) RoundTrip(req *http.Request) (*http.Response, error)
 	if etag != "" {
 		resp.Header.Set("X-Conditional-Request", etag)
 	}
+
+	githubTokenMetrics(resp.Header, reqStartTime)
+	ghRequestsGauge.With(prometheus.Labels{"path": req.URL.Path, "status": string(resp.StatusCode), "duration": roundTripTime.String()}).Inc()
```

Comment:
The path is _not_ appropriate here. Many times the path contains data about the request. For one example of many, look at the path for [getting repo data](https://developer.github.com/v3/repos/#get):

```
GET /repos/:owner/:repo
```

We want to track the fact that a repo's data is being fetched, but not necessarily which repo's data we fetch the most. Unbounded use of labels like this will also balloon the size of the metrics and quickly grow past the amount of data that can be scraped by Prometheus in one pass, leading to metrics collection failure.

Reply:
Would you say a cutoff after `/repos` for e.g. in the mentioned case would be appropriate?

Or would you suggest to mask the data e.g. `/repos/owner/reponame`. To collect all of these request under one.

Reply:
I've gone ahead and implemented a change for returning only the first fragment of the path. This will allow to distinguish between all major API calls referenced in the schema: https://api.github.com/

Reply:
That fails to distinguish 

```
GET /repos/:owner/:repo/branches/:branch
```

As documented [here](https://developer.github.com/v3/repos/branches/#get-branch)

Reply:
So while it gives us _some_ utility it is not a lot.